src/src/cron/usercreation.service.ts

```diff
@@ -0,0 +1,117 @@
+import { Injectable } from '@nestjs/common';
+import { Cron } from '@nestjs/schedule';
+import { ConfigService } from '@nestjs/config';
+import { AwsRekognitionService } from '../services/aws-rekognition/aws-rekognition.service';
+import { HasuraService } from '../services/hasura/hasura.service';
+
+@Injectable()
+export class UserCreationService {
+	private data_limit: string;
+
+	constructor(
+		private configService: ConfigService,
+		private awsRekognitionService: AwsRekognitionService,
+		private hasuraService: HasuraService,
+	) {
+		this.data_limit = this.configService.get<string>(
+			'AWS_REKOGNITION_INDEX_USER_BATCH_SIZE',
+		);
+	}
+
+	//first cron runs for each hour's 5th minute eg: 10:05am, 11::05am
+	@Cron('05 * * * *')
+	async createCollectionUsers() {
+		try {
+			/*----------------------- Create users in collection -----------------------*/
+			console.log(
+				'cron job 1: createCollectionUsers started at time ' +
+					new Date(),
+			);
+			const collectionId = this.configService.get<string>(
+				'AWS_REKOGNITION_COLLECTION_ID',
+			);
+			//delete collection if required
+			/*await this.awsRekognitionService.deleteCollection(
+				collectionId,
+				'userId',
+				'faceId',
+			);*/
+			//Step-1: Create collection if not exists
+			await this.awsRekognitionService.createCollectionIfNotExists(
+				collectionId,
+			);
+			// Step-2: Fetch all userIds which not added in collection
+			const nonCreatedUsers = await this.fetchAllUsersExceptCreated(
+				this.data_limit,
+			);
+			//console.log('>>>fetchAllUsersExceptCreated', nonCreatedUsers);
+			let nonexistusers = nonCreatedUsers.map((userObj) =>
+				String(userObj.id),
+			);
+			//console.log('>>>nonexistusers', nonexistusers);
+			// Step-3: Create not created users in collection and update status in users table
+			await this.awsRekognitionService.createUsersInCollection(
+				collectionId,
+				nonexistusers,
+			);
+		} catch (error) {
+			console.log(
+				'Error occurred in createCollectionUsers.',
+				error,
+				error.stack,
+			);
+		}
+	}
+	async fetchAllUsersExceptCreated(limit) {
```

Comment:
add empty line before starting new function

Reply:
i added empty line in all function starting